From f43f6d312b52702ce8020477960ab0521018b745 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Tue, 30 Jul 2013 12:16:44 +0000
Subject: [PATCH] abuild: fix ldpath handling

it wrongly scanned /lib instead of $datadir/lib, causing bad things to
happen when scanning provides
---
 abuild.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/abuild.in b/abuild.in
index 8c93658..51b4a6a 100755
--- a/abuild.in
+++ b/abuild.in
@@ -1020,7 +1020,7 @@ find_scanelf_paths() {
 	local datadir="$1"
 	local paths="$datadir/lib:$datadir/usr/lib" i= rpaths=
 	if [ -n "$ldpath" ]; then
-		paths="$paths:$(echo "$ldpath" | sed "s|\(^\|:\)|\1$datadir|g")"
+		paths="$paths:$(echo "${datadir}${ldpath}" | sed "s|:|:$datadir|g")"
 	fi
 	# search in all rpaths
 	for rpaths in "$pkgbasedir"/.control.*/.rpaths; do
-- 
1.8.3.4

